src/sentry/testutils/silo.py

```diff
@@ -164,3 +166,69 @@ def test_something(self):
     """
     with override_settings(SILO_MODE=SiloMode.MONOLITH):
         yield
+
+
+def psql_function(
+    name: str,
+    body: str,
+    params: Iterable[Tuple[str, str]] = tuple(),
+    pg_locals: Iterable[Tuple[str, str]] = tuple(),
+    return_type="text",
+) -> str:
+    params_encoded = ", ".join(" ".join(t) for t in params)
+    pg_locals_encoded = "; ".join(" ".join(t) for t in pg_locals)
+    return f"create or replace function {name}({params_encoded}) returns text as $$ declare {pg_locals_encoded} begin {body} end $$ language 'plpsql'"
+
+
+duplicate_role_body = """
+EXECUTE 'CREATE ROLE ' || to_name;
+FOR info IN
+    select * from information_schema.table_privileges where table_schema='public' and grantee = 'A'
+LOOP
+    /*this is the main statement to grant any privilege*/
+    execute 'GRANT '|| info.privilege_type ||' on table public.'|| info.table_name || ' to ' || from_name;
+END LOOP;
+return '';
+"""
+
+
+# Only use this is in tests -- it's not safe to allow in any production use case.
+def duplicate_user(from_user: str, to_user: str, connection: Any):
```

Comment:
Should we add the limited access role during the database bootstrapping that is done in `make create-db`

Reply:
I came up with a more clever solution that prevents the need for doing so.  Mostly, I don't want this ghost role in production or self hosted databases -- I've made an executive decision NOT to roll this out outside of tests.